### PR TITLE
RFC 2: Design Considerations for the Mediachain at Scale

### PR DESCRIPTION
**Pull request by vyzo**
Changes since last review: added Fault Tolerance section, fixed issues pointed by @parkan

### CODE REVIEW

File: rfc/mediachain-rfc-2.md [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+through multiple paths. As such, it can withstand a number of
+nodes trying to deny or disrupt the service by suppressing
+broadcast messages. 
+
+At worst, clients of adversarial nodes will be denied service.
+Affected clients can easily recover by connecting to another Peer for
+service. If too many nodes are dropping transactions so as to affect
+the system as a whole, the disruption would be big enough for a metric
+event visible to system monitoring. At which point, intervention
+by tier-1 node operators can restore the problem,.
```
Comment (parkan) [inline rfc/mediachain-rfc-2.md]: extra "," here

File: rfc/mediachain-rfc-2.md [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+
+The network is open to all clients, thus we are more concerned
+about malicious behavior than genuine bugs in client activity.
+
+A client can attempt a poisoning attack by injecting invalid
+transactions. Such transactions may refer to records not written
+to the datastore or be duplicates of older transactions, etc.
+The system can be rendered robust against transaction poisoning
+attacks by means of transaction verification. All peers verify
+client transactions before broadcasting them to the core network.
```
Comment (parkan) [inline rfc/mediachain-rfc-2.md]: Can you clarify this? Do you mean the Peer verification process under `Extending the Blockchain`? I like consistently using capitalized `Peer` to mean tier-1 nodes

File: rfc/mediachain-rfc-2.md (hunk 1 above)
Comment (vyzo) [inline rfc/mediachain-rfc-2.md]: fixed

File: rfc/mediachain-rfc-2.md (hunk 2 above)
Comment (vyzo) [inline rfc/mediachain-rfc-2.md]: I will capitalize accordingly -- although I tend to use small for the plural version.
Yes, the verification of the transaction is exactly the same as the one in Extending the Blockchain.

**Comment by vyzo** [comment]
Merging per our review on #tech

**Comment by parkan** [comment]
One more small point: in the [scaling limits](https://github.com/mediachain/mediachain/blob/master/rfc/mediachain-rfc-2.md#scaling-limits) section, we refer to the "index" and the "blockchain", but we don't really use the term "index" . Can you clarify what that means in terms of the journal? Do you mean the chain of just the `Block` structures without the referenced `JournalEntry`s?

**Comment by vyzo** [comment]
The index is the data structure mapping canonicals to chain head; the blockchain is the entire journal.
I'll add some clarification wording.

**Comment by vyzo** [comment]
Added wording in head commit.

**Pull request merged**